cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -201,6 +214,13 @@ runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
       Aeson.Object $ obj <> HMS.fromList [name .= Aeson.Null]
     toObject _ _ _ = Aeson.Null
 
+    toJsonPastHorizonException :: Qry.PastHorizonException -> Aeson.Value
```

Comment:
We should propagate this through the existing error type `ShelleyQueryCmdError`. I think it would be strange to render some errors via `ShelleyQueryCmdError` and some via JSON output.

Reply:
It depends.  If we go via `ShelleyQueryCmdError` if something wrong happens we get no values.  If we do this instead, we will get partial data even when some queries fail.

Reply:
I've taken the approach where if the epoch query fails, the JSON will be printed without the epoch and an error is logged to `stdout`.